madmom/ml/crf.py

```diff
@@ -0,0 +1,76 @@
+# encoding: utf-8
+"""
+This module contains an implementation of Conditional Random Fields (CRFs)
+"""
+import numpy as np
+
+from ..processors import Processor
+
+
+class ConditionalRandomField(Processor):
+    """
+    Implements a linear-chain Conditional Random Field using a
+    matrix-based definition:
+
+    P(Y|X) = exp(E(Y,X)) / Σ_Y'[E(Y', X)]
```

Comment:
Unfortunately this do not render nicely with Sphinx, try inserting `.. math::` above.